sqlx-core/src/postgres/options/mod.rs

```diff
@@ -89,7 +89,33 @@ pub struct PgConnectOptions {
 
 impl Default for PgConnectOptions {
     fn default() -> Self {
-        Self::new()
+        let port = var("PGPORT")
+            .ok()
+            .and_then(|v| v.parse().ok())
+            .unwrap_or(5432);
+
+        let host = var("PGHOST").ok().unwrap_or_else(|| default_host(port));
+
+        let username = var("PGUSER").ok().unwrap_or_else(whoami::username);
+
+        let database = var("PGDATABASE").ok();
+
+        PgConnectOptions {
+            port,
+            host,
+            socket: None,
+            username,
+            password: var("PGPASSWORD").ok(),
+            database,
+            ssl_root_cert: var("PGSSLROOTCERT").ok().map(CertificateInput::from),
+            ssl_mode: var("PGSSLMODE")
+                .ok()
+                .and_then(|v| v.parse().ok())
+                .unwrap_or_default(),
+            statement_cache_capacity: 100,
+            application_name: var("PGAPPNAME").ok(),
+            log_settings: Default::default(),
+        }
```

Comment:
This change causes `Self::default()` to ignore the `.pgpass` file entirely which I'm not sure was intentional.

It may be better to add a private static method `new_without_pgpass()` or something and move all this code there, and then use that in the `FromStr` impl and then have `new()` and `default()` return `Self::new_without_pgpass().apply_pgpass()`.

Reply:
Ah yes, that sounds nicely semantic. So the current default would become `new_without_pgpass`?

Reply:
`default()` and `new()` would both do `Self::new_without_pgpass().apply_pgpass()` so that they retain their current semantics.

Then in the `FromStr` impl you would use `Self::new_without_pgpass()` _instead_ of `Self::default()` so that the `.pgpass` file parsing doesn't get applied prematurely.

Reply:
@D1plo1d do you have time to make the changes suggested here?

Reply:
@abonander thanks for the ping - I just pushed a commit with the changes you suggested.